tonic/src/transport/server/mod.rs

```diff
@@ -846,3 +877,55 @@ where
         future::ready(Ok(svc))
     }
 }
+
+// TODO: handle this error
+async fn rdma_serve(addr: SocketAddr, routes: Routes) -> Result<(), io::Error> {
+    let rdma = Arc::new(
+        RdmaBuilder::default()
+            .set_max_message_length(MAX_MSG_LEN)
+            .listen(addr)
+            .await?,
+    );
+    let routes_clone = routes.clone();
+    tokio::spawn(rdma_serve_inner(Arc::clone(&rdma), routes_clone));
+    loop {
+        let rdma = rdma.listen().await?;
+        let routes = routes.clone();
+        tokio::spawn(async move { rdma_serve_inner(Arc::new(rdma), routes).await });
+    }
+}
+
+const MAX_MSG_LEN: usize = 10240;
+
+// TODO: handle this error
+async fn rdma_serve_inner(rdma: Arc<Rdma>, mut routes: Routes) -> Result<(), io::Error> {
+    println!("[server] connected!");
+
+    loop {
+        let (req_mr, len) = rdma.receive_with_imm().await?;
```

Comment:
Use RDMA SEND/RECV operations to implement RPC is a good start, and we can think about how to use RDMA WRITE or READ operations to improve performance.